### PR TITLE
solution

### PR DESCRIPTION
**Pull request by crash2bash**
[DEMO LINK](https://crash2bash.github.io/react_todo-app/)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,85 +1,181 @@
...
-            </div>
-            <input type="text" className="edit" />
-          </li>
-        </ul>
-      </section>
+const filterTypes = ['All', 'Active', 'Completed'];
+
+const filterTodos = (todos, filter) => {
+  if (filter === 'Active') {
+    return todos.filter(todo => todo.completed === false);
```
Comment (irenhh) [inline src/App.js:10]:
```suggestion
    return todos.filter(todo => !todo.completed);
```

File: src/App.js [hunk 2]
```diff
@@ -1,85 +1,181 @@
...
+  }
+
+  toggledCheck = (id, checked) => {
+    const { filter } = this.state;
+
+    this.setState((prevState) => {
+      const allTodos = prevState.todos.map(todo => (
+        todo.id === id
+          ? {
+            ...todo, completed: checked,
```
Comment (irenhh) [inline src/App.js:65]:
```suggestion
            ...todo, 
            completed: checked,
```

File: src/App.js [hunk 3]
```diff
@@ -1,85 +1,181 @@
...
-          </li>
-        </ul>
-
-        <button type="button" className="clear-completed">
-          Clear completed
-        </button>
-      </footer>
-    </section>
-  );
+  filtered = (evt) => {
```
Comment (irenhh) [inline src/App.js:77]: filtered what? functions must have names with verb (infinitive)

File: src/App.js [hunk 4]
```diff
@@ -1,85 +1,181 @@
...
+    const filter = evt.target.getAttribute('data-filter');
```
Comment (irenhh) [inline src/App.js:78]: don't interact with dom directly, use arguments for function

File: src/components/TodosFilter.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    ))}
+  </ul>
+);
+
+TodosFilter.propTypes = {
+  filterTypes: PropTypes.arrayOf(
+    PropTypes.string.isRequired,
+  ).isRequired,
+  filtered: PropTypes.func.isRequired,
+  filter: PropTypes.oneOfType([
```
Comment (irenhh) [inline src/components/TodosFilter.js:29]: why the types are different?

File: src/App.js [hunk 6]
```diff
@@ -1,85 +1,181 @@
...
-          </li>
-        </ul>
-
-        <button type="button" className="clear-completed">
-          Clear completed
-        </button>
-      </footer>
-    </section>
-  );
+  filteredItems = (filter) => {
```
[the remark this answers is not in the document]
Comment (crash2bash) [reply src/App.js:60]: You right, thanks! :)

File: src/App.js [hunk 7]
```diff
@@ -1,85 +1,181 @@
...
-      </section>
+const filterTypes = ['All', 'Active', 'Completed'];
+
+const filterTodos = (todos, filter) => {
+  if (filter === 'Active') {
+    return todos.filter(todo => !todo.completed);
+  }
+
+  if (filter === 'Completed') {
+    return todos.filter(todo => todo.completed === true);
```
[the remark this answers is not in the document]
Comment (crash2bash) [reply src/App.js:14]: Просто человек, у которого ты проверяешь эту работу, - балбес))) и забыл поубирать лишнее в этом блочке))

**Comment by irenhh** [comment]
almost there :)

since it's a project for portfolio, it must be perfect so that there is a couple of comments more

image
use styles that you already have in css files for selected tab and add cursor pointer

**Pull request closed without merging**